linkml/generators/sqltablegen.py

```diff
@@ -335,6 +358,42 @@ def get_id_or_key(cn: str, sv: SchemaView) -> str:
         pk_name = pk.alias if pk.alias else pk.name
         return f"{cn}.{pk_name}"
 
+    # Takes in the existing autogenerated list, the column list, and an annotation to extract the index
+    def parse_annotation_indexes(self, autogen: list, cols: list, ann) -> list:
+        index_list = []
+        value_dict = {k: annotation for k, annotation in ann.value._items()}
+        for key, value in value_dict.items():
+            name_exists = self.check_duplicate_entry_names(autogen, cols, key)
+            if not name_exists:
+                annotated_ind = Index(key, *value)
+                index_list.append(annotated_ind)
+        return index_list
+
+    # Scans the schema for duplicate names that exist as a best practice
+    @staticmethod
+    def check_duplicate_entry_names(autogen: list, cols_list: list, item_name: str) -> bool:
+        for entry in cols_list:
+            col_entry_name = getattr(entry, "name", str(entry))
+            if isinstance(entry, Index) or isinstance(entry, UniqueConstraint):
+                if item_name == col_entry_name:
+                    exp_list = [getattr(col, "name", str(col)) for col in entry.expressions]
+                    msg = (
+                        "Warning: proposed item name already exists in Table"
+                        "Please generate a new name: "
+                        f"{item_name} already exists for {col_entry_name}, with columns {str(exp_list)}"
+                    )
+                    logger.info(msg)
+                    return True
+        if item_name in autogen:
+            msg = (
+                "Warning: proposed item name already exists in schema"
+                "Please generate a new name: "
+                f"{item_name} already exists"
+            )
+            logger.info(msg)
+            return True
+        return False
```

Comment:
Do your tests cover any cases where this function returns True?

If not, you should add extra tests to do so.